### PR TITLE
Upgrades `kube-rs` to 0.75 and `k8s-openapi` to 0.16

### PR DESCRIPTION
**Pull request by jpmcb**
```
Ignores duplicate dependencies in the integration tests which are brought in via the aws config / sdk

Signed-off-by: John McBride <jpmmcb@amazon.com>
```

**Issue number:**

This will close a few stale dependabot PRs automatically

**Description of changes:**

- Upgrades `kube-rs` to 0.75 and `k8s-openapi` to 0.16. Upgrades a few methods to non-deprecated APIs in `kube`:

```
Deprecated since 0.74.0: 

ResourceExt::name can panic and has been replaced by ResourceExt::name_any and ResourceExt::name_unchecked.
This fn will be removed in 0.77.0.
```

[Reference](https://docs.rs/kube/latest/kube/trait.ResourceExt.html#tymethod.name)

```
Deprecated since 0.72.0: 

fn replaced with the WatchStreamExt::touched_objects which can be chained onto watcher.
Add use kube::runtime::WatchStreamExt; and call stream.touched_objects() instead.
This function will be removed in 0.75.0.
```

[Reference](https://docs.rs/kube/latest/kube/runtime/utils/fn.try_flatten_touched.html)

- Also, since the integration tests is the only crate that's pulling in AWS dependencies, we can safely ignore the duplicate dependencies causing problems through the `aws-smithy-client` tree.

- Finally, this patch also removes a few const variables that weren't being used and were throwing warnings during compile time
 
**Testing done:**

_Coming soon!_

Also, still need to verify that this works with creating IPv6 clusters (since there were comments about `rustls-tls` being tied in with kube 0.71 - @gthao313, do you have any insight on this? Or with how to test IPv6 clusters?)

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

File: deny.toml [hunk 1]
```diff
@@ -43,7 +43,9 @@ skip-tree = [
     { name = "structopt-derive", version = "0.4.18"},
     # use "rustls-tls" feature in kube to build correct k8s client for IPv6 cluster
     # with "rustls-tls" enabled, kube uses mixed versions of hyper-rustls
-    { name = "kube", version = "0.71.0"}
+    { name = "kube", version = "0.75.0"},
+    # aws-smithy-client
```
Comment (jpmcb) [inline deny.toml]: Need to add a better comment here

File: deny.toml [hunk 2]
```diff
@@ -43,7 +43,9 @@ skip-tree = [
     { name = "structopt-derive", version = "0.4.18"},
     # use "rustls-tls" feature in kube to build correct k8s client for IPv6 cluster
     # with "rustls-tls" enabled, kube uses mixed versions of hyper-rustls
```
Comment (jpmcb) [inline deny.toml:45]: Need to reconcile these comments - what's the state of `rustls-tls` in kube 0.75?

File: agent/Cargo.toml [hunk 3]
```diff
@@ -17,9 +17,9 @@ tracing-subscriber = { version = "0.3", features = ["registry", "env-filter"] }
 tracing-opentelemetry = "0.16"
 
 # k8s-openapi must match the version required by kube and enable a k8s version feature
-k8s-openapi = { version = "0.14.0", default-features = false, features = ["v1_20"] }
+k8s-openapi = { version = "0.16.0", default-features = false, features = ["v1_20"] }
 # "rustls-tls" feature must be used kube 0.71.0 to build correct k8s client for IPv6 cluster
```
Comment (jpmcb) [inline agent/Cargo.toml]: Need to reconcile this comment

File: apiserver/Cargo.toml [hunk 4]
```diff
@@ -26,9 +26,9 @@ tracing-subscriber = { version = "0.3", features = ["registry", "env-filter"] }
 tracing-opentelemetry = "0.16"
 
 # k8s-openapi must match the version required by kube and enable a k8s version feature
-k8s-openapi = { version = "0.14.0", default-features = false, features = ["v1_20"] }
+k8s-openapi = { version = "0.16.0", default-features = false, features = ["v1_20"] }
 # "rustls-tls" feature must be used kube 0.71.0 to build correct k8s client for IPv6 cluster
```
Comment (jpmcb) [inline apiserver/Cargo.toml]: Need to reconcile this comment

**Review by webern — APPROVED** [review]
looks good pending testing and the reconciliation of that rustls comment

File: agent/Cargo.toml (hunk 3 above)
Comment (gthao313) [reply agent/Cargo.toml]: We're safe to remove this comment. Kube 0.75.0 won't break IPV6 cluster.

**Review by gthao313 — APPROVED** [review]
IPV6 cluster test passed! :nice:

```
brs: "brs-ip-192-168-133-156.us-west-2.compute.internal"      current_version: "1.10.0"       current_state: Idle
brs: "brs-ip-192-168-138-25.us-west-2.compute.internal"      current_version: "1.10.0"       current_state: Idle
brs: "brs-ip-192-168-140-119.us-west-2.compute.internal"      current_version: "1.10.0"       current_state: Idle
[Complete]: All nodes has been successfully updated to latest version!
```

**Comment by jpmcb** [comment]
Force pushed to address `rust-tls` comments - will merge once checks pass
